larky/src/test/java/com/verygood/security/larky/modules/vgs/vault/VaultModuleSPITest.java

```diff
@@ -0,0 +1,125 @@
+package com.verygood.security.larky.modules.vgs.vault;
+
+import com.verygood.security.larky.console.testing.TestingConsole;
+import com.verygood.security.larky.modules.VaultModule;
+import net.starlark.java.eval.EvalException;
+import net.starlark.java.eval.Starlark;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+// Tests that VaultModule SPI functionality works as expected
```

Comment:
what's SPI?

Reply:
It's the Java Service Module Interface. I'm using it to implicitly detect, at runtime, the VaultModule Implementation to be used (instead of doing it explicitly with `addOverride` function). This is how I use it: https://github.com/verygoodsecurity/starlarky/pull/109/files#diff-cf5faff91ae9cd74b7b9d2504673969166da7d94e0b519c53f7b0dfaa2fd856fR32